src/hooks/Guilds/guild/useProposalCalls.ts

```diff
@@ -70,26 +69,30 @@ const useProposalCalls = (guildId: string, proposalId: string) => {
           const filteredActions = calls.filter(
             call => call.data !== ZERO_HASH || !call.value?.isZero()
           );
-
-          const actions = await Promise.all(
-            filteredActions.map(async (call, index, allCalls) => {
-              if (isApprovalCall(call)) {
-                const { decodedCall } = await decodeCall(
-                  call,
-                  contracts,
-                  chain?.id
-                );
-                allCalls[index + 1] = {
-                  ...allCalls[index + 1],
-                  approval: {
-                    amount: decodedCall?.args?._value,
-                    token: call.to,
-                  },
-                };
-              }
-              return call;
-            })
-          );
+          const actions = [];
+          for (const callIndex in filteredActions) {

```

Comment:
This is just me nitpicking, so you can disregard this, but `for...in`-loop has been considered bad practice since it also iterates over the prototype chain. It could potentially cause performance issues if we have a lot of props to loop over. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in

Reply:
Same as before. Was in draft and was working on that. I removed this! But good comment =)